src/Util/Datastore.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace App\Util;
+
+use League\Flysystem\FileExistsException;
+use League\Flysystem\FilesystemInterface;
+
+use Psr\Log\LoggerInterface;
+use Ramsey\Uuid\Uuid;
+
+/**
+ * Datastore utility class which manipulates files on disk.
+ */
+class Datastore
+{
+    /**
+     * Flysystem interface object for Datastore.
+     *
+     * @var FilesystemInterface
+     */
+    private $datastoreFlysystem;
+
+    /**
+     * Relative path for files folder.
+     */
+    const FILES_DIRECTORY = 'files';
+
+    /**
+     * Logger interface instance for Monolog default channel.
+     *
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * Datastore constructor.
+     *
+     * @param FilesystemInterface $datastoreFlysystem
+     */
+    public function __construct(FilesystemInterface $datastoreFlysystem)
+    {
+        $this->datastoreFlysystem = $datastoreFlysystem;
+    }
+
+    /**
+     * Setting the logger interface.
+     *
+     * @param LoggerInterface $logger Monolog logger interface instance.
+     *
+     * @return void
+     */
+    public function setLogger(LoggerInterface $logger): void
```

Comment:
This is not used. Should this be in the contructor?

Reply:
We discussed this on Friday, we decided we will use the default monologger https://symfonycasts.com/screencast/symfony4-fundamentals/logger-trait

Reply:
Ah ok. Isn't this the old method? Instead you want to implement loggerInterface or something? Or just add it to the constructor, it should use the default logger.

Reply:
LoggerTrait was creating another Helper class etc, and the other method of using it in the constructor was the usual way and we thought we will do it like this because we don't have to specify any specific channel and it goes to the default

Reply:
Moved it to the constructor, but I thought this was the way to inject default channel